alerting/src/test/kotlin/org/opensearch/alerting/MonitorDataSourcesIT.kt

```diff
@@ -755,6 +756,59 @@ class MonitorDataSourcesIT : AlertingSingleNodeTestCase() {
         assertEquals("Didn't match all 4 queries", 1, findings[0].docLevelQueries.size)
     }
 
+    fun `test cleanup monitor on partial create monitor failure`() {
+        val docQuery = DocLevelQuery(query = "dnbkjndsfkjbnds:\"us-west-2\"", name = "3")
+        val docLevelInput = DocLevelMonitorInput("description", listOf(index), listOf(docQuery))
+        val trigger = randomDocumentLevelTrigger(condition = ALWAYS_RUN)
+        val customQueryIndex = "custom_alerts_index"
+        val analyzer = "dfbdfbafd"
+        val testDoc = """{
+            "rule": {"title": "some_title"},
+            "message": "msg 1 2 3 4"
+        }"""
+        indexDoc(index, "2", testDoc)
+        client().admin().indices()
+            .create(
+                CreateIndexRequest(customQueryIndex + "-000001").alias(Alias(customQueryIndex))
+                    .mapping(
+                        """
+                        {
+                          "_meta": {
+                            "schema_version": 1
+                          },
+                          "properties": {
+                            "query": {
+                              "type": "percolator_ext"
+                            },
+                            "monitor_id": {
+                              "type": "text"
+                            },
+                            "index": {
+                              "type": "text"
+                            }
+                          }
+                        }
+                        """.trimIndent()
+                    )
+            ).get()
+
+        client().admin().indices().close(CloseIndexRequest(customQueryIndex + "-000001")).get()
+        var monitor = randomDocumentLevelMonitor(
+            inputs = listOf(docLevelInput),
+            triggers = listOf(trigger),
+            dataSources = DataSources(
+                queryIndex = customQueryIndex,
+                queryIndexMappingsByType = mapOf(Pair("text", mapOf(Pair("analyzer", analyzer)))),
+            )
+        )
+        try {
+            createMonitor(monitor)
+            fail("monitor creation should fail due to incorrect analyzer name in test setup")
+        } catch (e: Exception) {
+            Assert.assertEquals(client().search(SearchRequest(SCHEDULED_JOBS_INDEX)).get().hits.hits.size, 0)
```

Comment:
Assert queryIndex has 0 docs and no new mappings applied

Reply:
We cant assert that because we dont fail delete monitor request when query index docs clean up fails.
So even though we call delete monitor there is no guarantee of that. It's only a best effort clean up